backend/src/models/user.js

```diff
@@ -1,35 +1,56 @@
-const mongoose = require('../libs/mongoose.js');
+const sequelize = require('../libs/sequelize.js');
+const Sequelize = require('sequelize');
 const crypto = require('crypto');
-
-const userSchema = new mongoose.Schema({
-    name: { type: String, required: true },
-    passwordHash: { type: String, required: true },
-    salt: { type: String, required: true },
-    created: { type: Date, default: Date.now },
-    token: { type: Date, default: Date.now }
-});
-
-userSchema.virtual('password')
-.set(function (password) {
-    this._plainPassword = password;
-    if (password) {
-        this.salt = crypto.randomBytes(128).toString('base64');
-        this.passwordHash = crypto.pbkdf2Sync(password, this.salt, 1, 128, 'sha1');
-    } else {
-        this.salt = undefined;
-        this.passwordHash = undefined;
-  }
-})
-.get(function () {
-    return this._plainPassword;
+const User = sequelize.define('user', {
+    ID : {
+        type: Sequelize.INTEGER,
+        autoIncrement: true,
+        primaryKey: true
+    },
+    email: {
+        type: Sequelize.STRING,
+        allowNull: false,
+        unique: true
+    },
+    username: {
+        type: Sequelize.STRING,
+        allowNull: false,
+        unique: true
+    },
+    passwordHash: {
+        type: Sequelize.TEXT,
+        allowNull: false
+    },
+    salt: {
+        type: Sequelize.STRING,
+        allowNull: false
+    },
+    token: {
+        type: Sequelize.STRING
```

Comment:
why do you keep token in the db model? what if user has several devices and several tokens for each of them ?